pkg/templates/clusterconfig_funcs.go

```diff
@@ -51,10 +56,10 @@ func (t *TemplateResolver) fromClusterClaim(claimname string) (string, error) {
 	}
 
 	if _, ok := spec["value"]; ok {
-		t.addToReferencedObjects(getObj.GetAPIVersion(), getObj.GetKind(), "", claimname)
-
 		return spec["value"].(string), nil
 	}
 
+	t.addToReferencedObjects(clusterClaimAPIVersion, "ClusterClaim", "", claimname)
```

Comment:
I am not sure about this ?  In the case of No Error/good result - this stmt wont be reached  and referenced objects wont be added , correct?
there is a return stmt above  it in the green path if\else

Reply:
Good catch! I fixed this in the latest push and also added a unit test.